akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorAdapter.scala

```diff
@@ -103,13 +106,15 @@ import scala.util.control.NonFatal
     handle(ctx.messageAdapters)
   }
 
-  private def withSafelyAdapted[U, V](adapt: () ⇒ U)(body: U ⇒ V): Unit =
-    try body(adapt())
-    catch {
-      case NonFatal(ex) ⇒
+  private def withSafelyAdapted[U, V](adapt: () ⇒ U)(body: U ⇒ V): Unit = {
+    Try(adapt()) match {
```

Comment:
I'd prefer to do this without allocations from `Try` but couldn't see a way since the `null` can't be used with such `U` types

Reply:
Why can't it, `AdaptMessage` and `AdaptWithRegisteredMessageAdapter` are both internal so couldn't we just add a `Null` bound to the type parameter there, and here?

Reply:
somewhat strange to enforce that since all other messages can be of type `Any`

Reply:
ok, I was just thinking they'd be boxed anyways, but I'm not very upset ;)